Changes:

1) Allow environment to set CXX.
2) Do not build for both i386 and x86_64 since MacPorts' universal
   configuration will specify which architectures to build for.

diff -ru Ice-3.5.0.orig/cpp/config/Make.rules.Darwin Ice-3.5.0/cpp/config/Make.rules.Darwin
--- Ice-3.5.0.orig/cpp/config/Make.rules.Darwin	2013-03-11 08:19:46.000000000 -0700
+++ Ice-3.5.0/cpp/config/Make.rules.Darwin	2013-03-11 21:30:00.269042661 -0700
@@ -11,17 +11,11 @@
 # This file is included by Make.rules when uname is Darwin.
 #
 
-CXX			= xcrun clang++
+CXX			?= xcrun clang++
 
 CXXFLAGS		= -Wall -Werror -D_REENTRANT
 
 ifeq ($(OPTIMIZE),yes)
-     #
-     # By default we build binaries with both architectures when optimization is enabled.
-     #
-     ifeq ($(CXXARCHFLAGS),)
-     	CXXARCHFLAGS	:= -arch i386 -arch x86_64
-     endif   
      CXXFLAGS		:= $(CXXARCHFLAGS) -O2 -DNDEBUG $(CXXFLAGS)
 else
      CXXFLAGS		:= $(CXXARCHFLAGS) -g $(CXXFLAGS)
@@ -72,7 +66,7 @@
 ICEUTIL_OS_LIBS         = -lpthread
 ICE_OS_LIBS             = -ldl
 
-PLATFORM_HAS_READLINE   := no
+PLATFORM_HAS_READLINE   := yes
 
 #
 # QT is used only for the deprecated IceGrid and IceStorm SQL plugins
